internal/pkg/skuba/deployments/ssh/kubernetes.go

```diff
@@ -127,3 +129,35 @@ func kubernetesWaitForKubelet(t *Target, data interface{}) error {
 	}
 	return errors.Wrap(err, fmt.Sprintf("Timed out waiting for %s", t.target.Nodename))
 }
+
+// with every node added to the cluster, it takes approx. 60s more for the new node to get ready
```

Comment:
This means that `skuba node join -r worker ...` on 250 node cluster can take ~4h? That would be a bug.

Reply:
it would be a bug that we can't fix in skuba, yes.

@drpaneas do you think this scales linearly? i haven't done much testing and counting time

Reply:
From my experience this does not scale linearly. This is log from 50-node vmware cluster:
```
16:51:51 states.go:40] === state skuba-update.start applied successfully ===
16:54:14 states.go:40] === state skuba-update.start applied successfully ===
16:56:28 states.go:40] === state skuba-update.start applied successfully ===
16:58:35 states.go:40] === state skuba-update.start applied successfully ===
17:00:51 states.go:40] === state skuba-update.start applied successfully ===
17:03:07 states.go:40] === state skuba-update.start applied successfully ===
17:05:23 states.go:40] === state skuba-update.start applied successfully ===
17:07:33 states.go:40] === state skuba-update.start applied successfully ===
17:09:44 states.go:40] === state skuba-update.start applied successfully ===
17:11:51 states.go:40] === state skuba-update.start applied successfully ===
17:13:53 states.go:40] === state skuba-update.start applied successfully ===
17:16:04 states.go:40] === state skuba-update.start applied successfully ===
17:18:08 states.go:40] === state skuba-update.start applied successfully ===
17:20:16 states.go:40] === state skuba-update.start applied successfully ===
17:22:19 states.go:40] === state skuba-update.start applied successfully ===
17:24:35 states.go:40] === state skuba-update.start applied successfully ===
17:26:44 states.go:40] === state skuba-update.start applied successfully ===
17:29:17 states.go:40] === state skuba-update.start applied successfully ===
17:31:38 states.go:40] === state skuba-update.start applied successfully ===
17:33:46 states.go:40] === state skuba-update.start applied successfully ===
17:35:54 states.go:40] === state skuba-update.start applied successfully ===
17:38:15 states.go:40] === state skuba-update.start applied successfully ===
17:40:31 states.go:40] === state skuba-update.start applied successfully ===
17:43:13 states.go:40] === state skuba-update.start applied successfully ===
17:45:38 states.go:40] === state skuba-update.start applied successfully ===
17:48:04 states.go:40] === state skuba-update.start applied successfully ===
17:50:11 states.go:40] === state skuba-update.start applied successfully ===
17:52:19 states.go:40] === state skuba-update.start applied successfully ===
17:54:33 states.go:40] === state skuba-update.start applied successfully ===
17:56:47 states.go:40] === state skuba-update.start applied successfully ===
17:58:57 states.go:40] === state skuba-update.start applied successfully ===
18:01:14 states.go:40] === state skuba-update.start applied successfully ===
18:03:36 states.go:40] === state skuba-update.start applied successfully ===
18:05:41 states.go:40] === state skuba-update.start applied successfully ===
18:07:54 states.go:40] === state skuba-update.start applied successfully ===
18:10:25 states.go:40] === state skuba-update.start applied successfully ===
18:12:39 states.go:40] === state skuba-update.start applied successfully ===
18:14:50 states.go:40] === state skuba-update.start applied successfully ===
18:17:21 states.go:40] === state skuba-update.start applied successfully ===
18:19:46 states.go:40] === state skuba-update.start applied successfully ===
18:22:03 states.go:40] === state skuba-update.start applied successfully ===
18:24:18 states.go:40] === state skuba-update.start applied successfully ===
18:26:39 states.go:40] === state skuba-update.start applied successfully ===
18:29:16 states.go:40] === state skuba-update.start applied successfully ===
18:31:39 states.go:40] === state skuba-update.start applied successfully ===
18:34:05 states.go:40] === state skuba-update.start applied successfully ===
18:36:36 states.go:40] === state skuba-update.start applied successfully ===
18:39:18 states.go:40] === state skuba-update.start applied successfully ===
18:41:49 states.go:40] === state skuba-update.start applied successfully ===
18:44:21 states.go:40] === state skuba-update.start applied successfully ===
18:46:46 states.go:40] === state skuba-update.start applied successfully ===
18:51:11 states.go:40] === state skuba-update.start applied successfully ===
18:56:09 states.go:40] === state skuba-update.start applied successfully === 
```

Reply:
It's not linear for sure. One question to this:

Did you added the nodes one by one? e.g. add one, wait to become ready. Then add another one. Wait to become ready.

or

add all nodes at ones and monitor when they will become ready?

Reply:
```bash
for ((i=0; i<${#IP_WORKERS[@]}; i++)); do
    node_join worker "${IP_WORKERS[$i]}"
done
kubectl wait --for=condition=ready nodes --all --timeout=5m
```

Reply:
I have changed the timeout to 5m now, thanks @kravciak for the insights